<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ModifyMovieDialog</class>
 <widget class="QDialog" name="ModifyMovieDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Modyfikuj film</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QWidget {
    background-color: #f8f9fa;
    font-family: 'Segoe UI', Arial, sans-serif;
}

QGroupBox {
    font-weight: bold;
    border: 2px solid #dee2e6;
    border-radius: 8px;
    margin-top: 1ex;
    padding-top: 10px;
    background-color: white;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 5px 0 5px;
    color: #495057;
}

QPushButton {
    background-color: #007bff;
    color: white;
    border: none;
    padding: 8px 16px;
    border-radius: 6px;
    font-weight: bold;
    min-width: 80px;
}

QPushButton:hover {
    background-color: #0056b3;
}

QPushButton:pressed {
    background-color: #004085;
}

QPushButton:disabled {
    background-color: #6c757d;
    color: #adb5bd;
}

QLineEdit {
    border: 2px solid #ced4da;
    border-radius: 6px;
    padding: 8px 12px;
    background-color: white;
    selection-background-color: #007bff;
}

QLineEdit:focus {
    border-color: #007bff;
    outline: none;
}

QComboBox {
    border: 2px solid #ced4da;
    border-radius: 6px;
    padding: 8px 12px;
    background-color: white;
    min-width: 120px;
}

QComboBox:focus {
    border-color: #007bff;
}

QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    width: 20px;
    border-left-width: 1px;
    border-left-color: #ced4da;
    border-left-style: solid;
    border-top-right-radius: 6px;
    border-bottom-right-radius: 6px;
}

QComboBox::down-arrow {
    image: url(:/icons/arrow-down.png);
    width: 12px;
    height: 12px;
}

QRadioButton {
    spacing: 8px;
    color: #495057;
    font-weight: normal;
}

QRadioButton::indicator {
    width: 18px;
    height: 18px;
}

QRadioButton::indicator:unchecked {
    border: 2px solid #ced4da;
    border-radius: 9px;
    background-color: white;
}

QRadioButton::indicator:checked {
    border: 2px solid #007bff;
    border-radius: 9px;
    background-color: #007bff;
}

QRadioButton::indicator:checked:hover {
    background-color: #0056b3;
    border-color: #0056b3;
}

QLabel {
    color: #495057;
    font-weight: normal;
}</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="mainLayout">
   <item>
    <widget class="QGroupBox" name="mainGroupBox">
     <property name="title">
      <string>Informacje o filmie</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Tytuł:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="titleLineEdit">
        <property name="placeholderText">
         <string>Wprowadź tytuł filmu</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Rok produkcji:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="yearSpinBox">
        <property name="minimum">
         <number>1900</number>
        </property>
        <property name="maximum">
         <number>2030</number>
        </property>
        <property name="value">
         <number>2024</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Długość:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="lengthSpinBox">
        <property name="suffix">
         <string> min</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="value">
         <number>120</number>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Gatunek:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QComboBox" name="genreComboBox">
        <item>
         <property name="text">
          <string>Adventure</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Action</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Comedy</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Drama</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Horror</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Romance</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Sci-Fi</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Thriller</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Historical</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Reżyser:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QComboBox" name="directorComboBox"/>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Opis:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QTextEdit" name="descriptionTextEdit">
        <property name="placeholderText">
         <string>Wprowadź opis filmu</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Ocena (1-10):</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <layout class="QHBoxLayout" name="ratingLayout">
        <item>
         <widget class="QSlider" name="ratingSlider">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>10</number>
          </property>
          <property name="value">
           <number>5</number>
          </property>
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="ratingLabel">
          <property name="text">
           <string>5</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Status:</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QCheckBox" name="watchedCheckBox">
        <property name="text">
         <string>Film został obejrzany</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="actorsGroupBox">
     <property name="title">
      <string>Aktorzy</string>
     </property>
     <layout class="QVBoxLayout" name="actorsMainLayout">
      <item>
       <widget class="QScrollArea" name="actorsScrollArea">
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="scrollAreaWidgetContents">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>450</width>
           <height>148</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="actorsLayout"/>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::StandardButton::Cancel|QDialogButtonBox::StandardButton::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ModifyMovieDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>248</x>
     <y>254</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ModifyMovieDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>248</x>
     <y>254</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ratingSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ratingLabel</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>248</x>
     <y>254</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
